### ISSUE TITLE
[Bug]: Defined Functions Fail on Updated configurations

### ISSUE THREAD

**Issue by jdhitsolutions** [mdgrs-mei/PowerShellRun#62]
### OS

- [X] Windows
- [ ] macOS
- [ ] Linux

### PSVersionTable

```powershell
Name                           Value
----                           -----
PSVersion                      7.4.5
PSEdition                      Core
GitCommitId                    7.4.5
OS                             Microsoft Windows 10.0.26100
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### What happened?

I think the following behavior is a bug. We need a PowerShell script to build our configuration. I have that which defines favorites, entry groups, theming, etc. But I think there is a bug with functions defined between the PSRunFunctionRegistration commands. The first time I launch PowerShellRun after running my configuration script, I see my function.

![image](https://github.com/user-attachments/assets/5633d176-d04a-44db-8115-031387c97642)

`LuckyNumber` is the function defined in the configuration script. However, I am likely to update the configuration script for some other reason. I re-run the configuration script and launch PowerShellRun, and the `LuckyNumber` function is 'gone`.

![image](https://github.com/user-attachments/assets/44b843e6-aa13-460e-ae2c-60d29a87f272)

The `Last Boot` item is script block that happens to be in the same entry group.

When I go back to PowerShell, the `LuckyNumber` function is still defined and I can run it. If I update the function in the configuration script, launch PowerShell Run, I still don't see the function., but I have the updated function in my session. I think your GlobalStore isn't updating the entry if it already exists and it should.

**Comment by mdgrs-mei** [comment]
Yes, it registers functions that are newly created between `Start/Stop-PSRunFunctionRegistration`. If there is a way to detect what functions are updated between them efficiently, we can fix this issue but is there any way?

So we want to detect the function names B and C in this example:

```powershell
function A {
}
function B {
    'Already in a session'
}

Start-PSRunFunctionRegistration
function B {
    'Updated'
}
function C {
}
Stop-PSRunFunctionRegistration
```

**Comment by jdhitsolutions** [comment]
Maybe we need to look at my related question about how to add an existing function to the configuration. How can I take a function loaded from my profile and make it a PowerShellRun function?

**Comment by mdgrs-mei** [comment]
This issue has been fixed by the PR #66.
